test/Target/rocdl.mlir

```diff
@@ -14,18 +14,18 @@ func @rocdl_special_regs() -> !llvm.i32 {
   %5 = rocdl.workgroup.id.y : !llvm.i32
   // CHECK: call i32 @llvm.amdgcn.workgroup.id.z()
   %6 = rocdl.workgroup.id.z : !llvm.i32
-  // CHECK: call i32 @__ockl_get_local_size(i32 0)
-  %7 = rocdl.workgroup.dim.x : !llvm.i32
-  // CHECK: call i32 @__ockl_get_local_size(i32 1)
-  %8 = rocdl.workgroup.dim.y : !llvm.i32
-  // CHECK: call i32 @__ockl_get_local_size(i32 2)
-  %9 = rocdl.workgroup.dim.z : !llvm.i32
-  // CHECK: call i32 @__ockl_get_global_size(i32 0)
-  %10 = rocdl.grid.dim.x : !llvm.i32
-  // CHECK: call i32 @__ockl_get_global_size(i32 1)
-  %11 = rocdl.grid.dim.y : !llvm.i32
-  // CHECK: call i32 @__ockl_get_global_size(i32 2)
-  %12 = rocdl.grid.dim.z : !llvm.i32
+  // CHECK: call i64 @__ockl_get_local_size(i32 0)
+  %7 = rocdl.workgroup.dim.x : !llvm.i64
+  // CHECK: call i64 @__ockl_get_local_size(i32 1)
+  %8 = rocdl.workgroup.dim.y : !llvm.i64
+  // CHECK: call i64 @__ockl_get_local_size(i32 2)
+  %9 = rocdl.workgroup.dim.z : !llvm.i64
+  // CHECK: call i64 @__ockl_get_global_size(i32 0)
+  %10 = rocdl.grid.dim.x : !llvm.i64
+  // CHECK: call i64 @__ockl_get_global_size(i32 1)
+  %11 = rocdl.grid.dim.y : !llvm.i64
+  // CHECK: call i64 @__ockl_get_global_size(i32 2)
+  %12 = rocdl.grid.dim.z : !llvm.i64
```

Comment:
This seems like the bug fix from ConvertToROCDLIR.cpp, please split it out in its own PR, it seems to me to be independent from the new features